rules/azurerm_resources_missing_prevent_destroy.go

```diff
@@ -0,0 +1,176 @@
+func (r *AzurermResourcesMissingPreventDestroyRule) Severity() tflint.Severity {
+	return tflint.WARNING
+}
+
+// Link returns the rule reference link
+func (r *AzurermResourcesMissingPreventDestroyRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks resources for missing lifecycle prevent_destroy configuration
+func (r *AzurermResourcesMissingPreventDestroyRule) Check(runner tflint.Runner) error {
+	config := azurermResourcePreventDestroyRuleConfig{}
+
+	if err := runner.DecodeRuleConfig(r.Name(), &config); err != nil {
+		return err
+	}
+
+	// Use default resources if none specified in config
+	resourcesToCheck := config.Resources
+	if len(resourcesToCheck) == 0 {
+		resourcesToCheck = defaultPreventDestroyResources
+	}
+
+	for _, resourceType := range resourcesToCheck {
+		// Skip this resource if its type is excluded in configuration
+		if stringInSlice(resourceType, config.Exclude) {
+			logger.Debug("excluding", "resourceType", resourceType)
+			continue
+		}
+
+		resources, err := runner.GetResourceContent(resourceType, &hclext.BodySchema{
+			Blocks: []hclext.BlockSchema{
+				{
+					Type: lifecycleBlockName,
+					Body: &hclext.BodySchema{
+						Attributes: []hclext.AttributeSchema{
+							{Name: preventDestroyAttrName},
+						},
+					},
+				},
+			},
+		}, nil)
+
+		if err != nil {
+			return err
+		}
+
+		for _, resource := range resources.Blocks {
+			logger.Debug("checking", "resource type", resource.Labels[0], "resource name", resource.Labels[1])
+
+			hasPreventDestroy := false
+			preventDestroyValue := false
+
+			// Check if lifecycle block exists
+			for _, block := range resource.Body.Blocks {
+				if block.Type == lifecycleBlockName {
+					// Check if prevent_destroy attribute exists in lifecycle block
+					if attribute, ok := block.Body.Attributes[preventDestroyAttrName]; ok {
+						logger.Debug("found prevent_destroy attribute", "resource type", resource.Labels[0], "resource name", resource.Labels[1])
+
+						err := runner.EvaluateExpr(attribute.Expr, func(val bool) error {
+							hasPreventDestroy = true
+							preventDestroyValue = val
+							return nil
+						}, &tflint.EvaluateExprOption{WantType: &cty.Bool})
+
+						if err != nil {
+							// If we can't evaluate the expression, assume it's present but might be dynamic
+							hasPreventDestroy = true
+							preventDestroyValue = true
+						}
+						break
+					}
+				}
+			}
+
+			// Emit issue if prevent_destroy is missing or set to false
+			if !hasPreventDestroy {
+				issue := "Resource is missing lifecycle { prevent_destroy = true }. This resource contains data that should be protected from accidental deletion."
+				if err := runner.EmitIssue(r, issue, resource.DefRange); err != nil {
+					return err
+				}
+			} else if !preventDestroyValue {
+				issue := "Resource has lifecycle { prevent_destroy = false }. Consider setting prevent_destroy = true to protect data from accidental deletion."
```

Comment:
I think it might be ok to allow it if it's explicitly set to `false`, but what do you think?
For example, this is intended for use in generating temporary resources. In my experience, there is a module that clones a database for testing purposes, and in that module `prevent_destroy` was set to `false`.

Or would it be more useful to be able to eliminate the possibility of it being set to false in all cases?